benchexec/tools/ultimate.py

```diff
@@ -204,113 +186,120 @@ def _requires_ultimate_data(self, executable):
         # all versions before 0.1.24 do not require ultimatedata
         return not (int(major) == 0 and int(minor) < 2 and int(patch) < 24)
 
-    def cmdline(self, executable, options, tasks, propertyfile=None, rlimits=None):
-        if rlimits is None:
-            rlimits = {}
-
-        self._uses_propertyfile = propertyfile is not None
-        if _OPTION_NO_WRAPPER in options:
-            # do not use old wrapper script even if property file is given
-            self._uses_propertyfile = False
-            propertyfile = None
-            options.remove(_OPTION_NO_WRAPPER)
+    def cmdline(self, executable, options, task, resource_limits):
+        combined_options = options + [*task.options] if task.options else []
```

Comment:
`task.options` is not a list of command-line parameters, but an arbitrary data structure (as defined in the task-definition file in the `options` key). So this needs to be converted to command-line parameters first. Cf. #646 and the `sv_benchmarks_util` that is added there for code that is useful for most cases. Maybe we should delay this PR until #646 is merged and then rebase on top of it?

Reply:
That sounds good! Lets do that.

Reply:
#646 is merged now.